exporters/opensearch-exporter/src/main/java/io/camunda/zeebe/exporter/opensearch/RestClientFactory.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.impl.nio.reactor.IOReactorConfig;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestClientBuilder;
+
+final class RestClientFactory {
+  private static final RestClientFactory INSTANCE = new RestClientFactory();
+
+  private RestClientFactory() {}
+
+  /**
+   * Returns a {@link RestClient} instance based on the given configuration. The URL is parsed as a
+   * comma separated list of "host:port" formatted strings. Authentication is supported only as
+   * basic auth; if there is no authentication present, then nothing is configured for it.
+   */
+  static RestClient of(final OpensearchExporterConfiguration config) {
+    return of(config, false);
+  }
+
+  /**
+   * Returns a {@link RestClient} instance based on the given configuration. The URL is parsed as a
+   * comma separated list of "host:port" formatted strings. Authentication is supported only as
+   * basic auth; if there is no authentication present, then nothing is configured for it.
+   *
+   * @param config the exporter configuration
+   * @param allowAllSelfSignedCertificates if set to true ALL self-signed certificates will be
+   *     accepted. This is meant for testing purposes only!
+   * @return the created {@link RestClient}
+   */
+  static RestClient of(
+      final OpensearchExporterConfiguration config, final boolean allowAllSelfSignedCertificates) {
+    return INSTANCE.createRestClient(config, allowAllSelfSignedCertificates);
+  }
+
+  private RestClient createRestClient(
+      final OpensearchExporterConfiguration config, final boolean allowAllSelfSignedCertificates) {
+    final HttpHost[] httpHosts = parseUrl(config);
+    final RestClientBuilder builder =
+        RestClient.builder(httpHosts)
+            .setRequestConfigCallback(
+                b ->
+                    b.setConnectTimeout(config.requestTimeoutMs)
+                        .setSocketTimeout(config.requestTimeoutMs))
+            .setHttpClientConfigCallback(
+                b -> configureHttpClient(config, b, allowAllSelfSignedCertificates));
+
+    return builder.build();
+  }
+
+  private HttpAsyncClientBuilder configureHttpClient(
+      final OpensearchExporterConfiguration config,
+      final HttpAsyncClientBuilder builder,
+      final boolean allowAllSelfSignedCertificates) {
+    // use single thread for rest client
+    builder.setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(1).build());
+
+    if (config.hasAuthenticationPresent()) {
+      setupBasicAuthentication(config, builder);
+    }
+
+    if (allowAllSelfSignedCertificates) {
+      // This code makes it so ALL self-signed certificates are accepted. This is meant for testing
+      // purposes only.
+      try {
+        final var sslcontext =
```

Comment:
:upside_down_face: Same here, we can use camelCase:

```suggestion
        final var sslContext =
```